server/channels/api4/report.go

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package api4
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/shared/mlog"
+)
+
+func (api *API) InitReports() {
+	api.BaseRoutes.Reports.Handle("/users", api.APISessionRequired(getUsersForReporting)).Methods("GET")
+}
+
+func getUsersForReporting(c *Context, w http.ResponseWriter, r *http.Request) {
+	if !(c.IsSystemAdmin() && c.App.SessionHasPermissionTo(*c.AppContext.Session(), model.PermissionSysconsoleReadUserManagementUsers)) {
```

Comment:
0/5 should this be and `&&` or an `||`. My guess is that sysadmins have always permissions for everything, so an `&&` here would be redundant. But I may be missing something regarding the permission schemes.

Reply:
It's just rearranging existing code, but I think it's okay to explicitly mention what access we need. thoughts @devinbinnie ?

Reply:
I'm not sure how this works explicitly, but I was under the impression that you can have an admin role without that permission, so we want to check for it. In any case I think most system admins do have it.